File: spec/python/test_opaque_external_type.py
import unittest
from opaque_external_type import OpaqueExternalType

class TestOpaqueExternalType(unittest.TestCase):
    def test_opaque_external_type(self):
        with OpaqueExternalType.from_file('src/term_strz.bin') as r:
            self.assertEqual(r.one.s1, "foo")
            self.assertEqual(r.one.s2, "bar")
            self.assertEqual(r.one.s3, "|baz@")

File: spec/lua/test_opaque_external_type.lua
local luaunit = require("luaunit")

require("opaque_external_type")

TestOpaqueExternalType = {}

function TestOpaqueExternalType:test_opaque_external_type()
    local r = OpaqueExternalType:from_file("src/term_strz.bin")

    luaunit.assertEquals(r.one.s1, "foo")
    luaunit.assertEquals(r.one.s2, "bar")
    luaunit.assertEquals(r.one.s3, "|baz@")
end
